.github/pull_request_template.md

```diff
@@ -0,0 +1,34 @@
+## Scope and purpose
+
+Fixes #<!-- ISSUE-ID -->. Dependent on #<!-- PULL-REQUEST-ID -->. <!-- Reference: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->
+
+Add a description of what this PR does and why it is needed. If a linked ticket(s) fully
+cover it, you can omit this.
+
+<!-- remove things that do not apply -->
+### This pull request
+* adds/changes/removes a dependency
+* changes the database
+* changes the API
+
+
+## Contributor Checklist
+
+Every pull request should have this checklist filled out, no matter how small it is.
+More information about contributing to NAV can be found in the
+[Hacker's guide to NAV](https://nav.readthedocs.io/en/latest/hacking/hacking.html#hacker-s-guide-to-nav).
+
+<!-- Add an "X" inside the brackets to confirm -->
+<!-- If not checking one or more of the boxes, please explain why below each. -->
+
+* [ ] Added a changelog fragment for [towncrier](https://nav.readthedocs.io/en/latest/hacking/hacking.html#adding-a-changelog-entry)
+* [ ] Added/amended tests for new/changed code
+* [ ] Added/changed documentation
+* [ ] Linted/formatted the code with black and ruff, easiest by using [pre-commit](https://nav.readthedocs.io/en/latest/hacking/hacking.html#pre-commit-hooks-and-black)
```

Comment:
We've been using a similar template in other projects for a while now, and I have to say I find this checkbox quite annoying by now.  The GitHub workflow will explicitly bonk me on the head and fail my PR if I haven't done this, so why do I need to check this box on every PR?

Reply:
It's mainly there so that new contributors aren't opening a pull request to then be surprised by the failing test, since they might not have read the `Hacking` docs. 

I personally find it satisfying to be able to check that checkbox every time, but it's open for discussion.

Reply:
There is still nothing but the Hacking docs that will prepare a new contributor for this.

A contributor can make huge changes with many commits, and still not be presented with this PR description template and this checkbox until they actually attempt to open a PR on GitHub - at which point they still have to go back and fix everything.

Reply:
Yes, that is true. So the question is if we want a new contributor to see that condition at the moment of creating a pull request or after opening it, when the relevant actions are failing.

Reply:
Its similar with the news fragment, a workflow will tell you that it wasn't included. But I do think the link to documentation for news fragments is very nice, same with the link to how to set up pre-commit.

Basically you will learn after your first pull request that you need to use pre-commit and make news fragments, and if you forget then a workflow will remind you, so I kinda agree that these dont need checkboxes

Reply:
I actually disagree stronger with the news fragment checkbox, because with the code if you once set up the pre-commit it is automatically fixed. On the other hand I had many times when I created a pull request that I was reminded of the news fragment by the PR template (in other repos) and I was happy that I could fix it before the check was failing

But I also just remembered that I added towncrier to pre-commit and it somehow does not stop me from committing things without a news fragment...Hmmm...Will have to check up on why that is

Reply:
Good point that you just set up pre-commit once then you're done while towncrier has to be added every time. 

I usually look at the checkboxes after opening the PR, since I don't then have to manually add X's, I can just click the boxes. And the towncrier workflow fails very fast, so it doesnt take much longer for the workflow to remind me I forgot it than it takes me to go through the checkboxes.

There are quite a few checkboxes here, so I think it could be a benefit to cut where possible so its less overwhelming. 

My opinions on this aren't very strong however, these are just my initial thoughts :stuck_out_tongue:

Reply:
Yes, I was thinking the same, so I'm considering just dropping that pre-commit hook. But then I would strongly prefer to have the check in the PR template